Framework/PythonInterface/plugins/algorithms/FindSXPeaksConvolve.py

```diff
@@ -247,5 +242,22 @@ def exec_child_alg(self, alg_name, **kwargs):
             return None
 
 
+def make_kernel(nrows, ncols, nbins):
+    # make a kernel with background subtraction shell of approx. same number of elements as peak region
+    kernel_shape, (nrows_bg, ncols_bg, nbins_bg) = get_kernel_shape(nrows, ncols, nbins)
+    kernel = np.zeros(kernel_shape)
+    kernel[nrows_bg:-nrows_bg, ncols_bg:-ncols_bg, nbins_bg:-nbins_bg] = 1
+    bg_mask = np.logical_not(kernel)
+    kernel[bg_mask] = -(nrows * ncols * nbins) / bg_mask.sum()  # such that kernel.sum() = 0
+    return kernel
+
+
+def get_kernel_shape(nrows, ncols, nbins):
+    nrows_bg = max(1, nrows // 16)  # padding either side of e.g. nrows for bg shell
+    ncols_bg = max(1, ncols // 16)
+    nbins_bg = max(1, nbins // 16)
+    return (nrows + 2 * nrows_bg, ncols + 2 * ncols_bg, nbins + 2 * nbins_bg), (nrows_bg, ncols_bg, nbins_bg)
```

Comment:
This is not obvious from the comment of the function. 
Can you add a general comment, like, ` #Padding edges of kernel for the background shell`